binary_search_tree/binary_search_tree.py

```diff
@@ -1,17 +1,69 @@
-class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+# class BinarySearchTree:
+#   def __init__(self, value):
+#     self.value = value
+#     self.left = None
+#     self.right = None
 
-  def insert(self, value):
-    pass
+#   def insert(self, value):
+#     pass
 
-  def contains(self, target):
-    pass
+#   def contains(self, target):
+#     pass
 
-  def get_max(self):
-    pass
+#   def get_max(self):
+#     pass
 
-  def for_each(self, cb):
-    pass
\ No newline at end of file
+#   def for_each(self, cb):
+#     pass
+
+class Node:
```

Comment:
well, at least you got a start on it! :D Keep at it, see if you can finish this on the weekend if you have the time!